src/cms/widgets/i18n.test.ts

```diff
@@ -0,0 +1,158 @@
+import test from 'tape'
+import {
+  getTranslationDefault,
+  upsertTranslation,
+  isTranslationCollection,
+  translate,
+  TranslationCollection,
+  hasTranslation,
+} from './i18n'
+
+const translations: TranslationCollection<string> = [
+  {
+    locale: 'de',
+    value: 'Katze',
+  },
+  {
+    locale: 'en',
+    value: 'cat',
+  },
+]
+test('hasTranslation', t => {
+  t.true(hasTranslation('de')(translations))
+  t.true(hasTranslation('en')(translations))
+  t.false(hasTranslation('es')(translations))
+  t.false(hasTranslation(null)(translations))
+  t.end()
+})
+
+test('getTranslationDefault', t => {
+  t.deepEquals(
+    getTranslationDefault('de')(translations),
+    'Katze',
+    'returns translation if existing',
+  )
+  t.equal(
+    getTranslationDefault('es')(translations),
+    undefined,
+    'returns undefined if non-existing',
+  )
+  t.equal(
+    getTranslationDefault('es', 'en')(translations),
```

Comment:
Maybe add the german translation as a second fallback value.

Reply:
Done